### DISCUSSION TITLE
Where to set a fallback Head title?

### DISCUSSION THREAD

**Discussion by florianwalther-private** [vercel/next.js#38228] [Help] [answered]
On some pages on my website, I want to have a specific `title` in the `<head>`. On all others, I want to use a default title. And I want pages that don't specify a `title` to fall back to this default title. Where do I set this default? Is it okay to set a title in `_app.tsx` and then set additional (more specific) titles on single pages? From inspecting the HTML it seems like it works but I want to make sure I'm not missing something.

`_app.tsx`:

``` 
  return (
    <SSRProvider>
      <div className={styles.appContainer}>
        <Head>
          <title>My fallback/default title</title>
        </Head>

        <NavBar
          [...]
        />
        <div className={styles.pageContainer}>
          <Component {...pageProps}
[...]
```

Page with more specific title:

```
return (
    <div>
      <Head>
        <title>Specific title that overrides the _app title</title>
      </Head>
```

Is this fine?

**Comment by icyJoseph** [comment, marked as answer]
It is fine. There's a package, one of the few I install every now and then, called, `next-seo`, which could help you with SEO related code. 

This package has the concept of `DefaultSeo`, which is placed at top of the `_app` as "fallback", or well, "default" SEO for pages.

IIRC the `Head` manager makes sure that there's no two title tags, so the last one kicks in.

> ⚠️ Make sure the fallback title goes before `<Component />` in `_app`! Otherwise, the fallback applies to every page!

You can do a little experiment:

```tsx
const Home: NextPage = () => {
  return (
    <div>
      <Head>
        <title>Home Page</title>
      </Head>
      <Head>
        <title>Home Page 2</title>
      </Head>

      Home
    </div>
  );
};

export default Home;
```

And then you'll see in the HTML, only one `title` is generated. And that'd be the second one. This is important for SEO, two titles might lead to unwanted results, in terms of SEO. 

This is also why the fallback title in `_app` has to go before `<Component />`

**Reply by florianwalther-private** [reply]
Thank you for clarifying! I checked the HTML myself and it seems fine. Just to clarify: This fallback feature works without the `next-seo` package you mentioned, right?

**Reply by icyJoseph** [reply]
Yes, this works out of the box.

It is often the case that people start to control open graph, twitter cards, facebook cards, etc, etc, and the meta tags SEO logic becomes overwhelming, then `next-seo` can come help us!

**Reply by florianwalther-private** [reply]
@icyJoseph Thank you very much!

**Reply by florianwalther-private** [reply]
@icyJoseph Should I use the `key` attribute on the `<title>` tags or is it not necessary? It was necessary for `og:title` to work properly.

**Reply by icyJoseph** [reply]
Interesting. How so? `key` is for component identity, at least React's key.

**Reply by florianwalther-private** [reply]
@icyJoseph It's actually mentioned in the docs here: https://nextjs.org/docs/api-reference/next/head

**Reply by icyJoseph** [reply]
Ah that's true. Interesting, I missed that. Good catch!

**Reply by florianwalther-private** [reply]
@icyJoseph The `<title>` tag seems to work without a key (meaning the _page_ one overrides the `_app` one) but for the `og:title` I had to use the key to get this behavior. I still add it to `<title>` now just to be safe.
